assets/js/googlesitekit/modules/datastore/modules.js

```diff
@@ -309,6 +378,54 @@ const baseResolvers = {
 			yield fetchGetModulesStore.actions.fetchGetModules();
 		}
 	},
+
+	*canActivateModule( slug ) {
+		const registry = yield Data.commonActions.getRegistry();
+		const module = registry.select( STORE_NAME ).getModule( slug );
+
+		// We use yield dataControls.resolveSelect in favour of registry.select
+		// to avoid a race condition where the module hasn't loaded yet.
+		// @TODO after #1769, we can do this.
+		// const module = yield dataControls.resolveSelect( STORE_NAME, 'getModule', slug );
+
+		// Return `undefined` if module with this slug isn't loaded yet.
+		if ( module === undefined ) {
+			return undefined;
+		}
+
+		// A module with this slug couldn't be found return null
+		if ( module === null ) {
+			return null;
+		}
+
+		const inactiveModules = [];
+
+		module.dependencies.forEach( ( dependencySlug ) => {
+			const dependedentModule = registry.select( STORE_NAME ).getModule( dependencySlug );
+			if ( ! dependedentModule.active ) {
+				inactiveModules.push( dependedentModule.name );
+			}
+		} );
+
+		// If we have inactive dependencies, there's no need to check if we can
+		// activate the module until the dependencies have been activated.
+		if ( inactiveModules.length ) {
+			const formatter = new Intl.ListFormat( getLocale(), { style: 'long', type: 'conjunction' } );
+
+			/* translators: Error message text. 1: A flattened list of module names. 2: A module name. */
+			const errorMessage = sprintf( __( 'You need to set up %1$s to gain access to %2$s.', 'google-site-kit' ), formatter.format( inactiveModules ), module.name );
+
+			yield baseActions.receiveCheckRequirementsError( { [ slug ]: errorMessage } );
```

Comment:
Apologies that this wasn't obvious from the ACs, but when we talk about "errors" here, this should be error **objects** like we use elsewhere in the plugin: Every error should be an object with (at minimum) `code` and `message` properties.

Also based on my above comment, let's change usage here to pass `slug` and `error` separately.